aips/aip-xx.md

```diff
@@ -0,0 +1,198 @@
+subscribe to multiple publishers. This will ensure that observers can continue to receive block data and events, even if
+some publishers go offline or network messages are lost in transit.
+
+## Reference Implementation
+
+A reference implementation of CO can be found in the
+[aptos-core repository](https://github.com/aptos-labs/aptos-core/tree/main/consensus/src/consensus_observer).
+
+## Testing
+
+Given the broad nature of this change, it is necessary to conduct extensive testing across all layers of the software
+stack. This includes: (i) individual unit tests for new components; (ii) integration tests for the consensus publisher
+and observer; (iii) end-to-end tests for network synchronization (e.g., via the existing smoke test infrastructure);
+(iv) performance tests to measure and verify the latency improvements of CO (e.g., via the existing performance test
+infrastructure in `forge`); and finally (v) pre-production tests in our `devnet` and `testnet` environments, before
+`mainnet` deployment.
+
+It is also particularly important to ensure that sufficient testing takes place for failure scenarios and edge cases.
+For example, node restarts, node failures, data wipes, invalid or malicious network messages, failures during 
+block execution and proof processing, etc. This will help to ensure CO is reliable enough to withstand real-world
+conditions.
+
+### Preliminary Performance Results
+
+Preliminary testing has already been conducted in several test environments, including `forge`. In these experiments,
+the performance of CO was measured and compared against state sync. The results of the experiments showed significant
+improvements in end-to-end transaction latencies for VFNs and PFNs. For example:
+1. In a semi-realistic, high-load scenario, CO reduced end-to-end transaction latencies for VFNs by `~30%`
+   (from `~2.1s` to `~1.5s`) at `~10k TPS`.
+2. In a medium-load scenario with extreme network chaos, CO reduced end-to-end transaction latencies for PFNs by
+   `~50%` (from `~3.1s` to `~1.6s`) at `~5k TPS`.
+
+## Risks and Drawbacks
+
+There are several risks and drawbacks associated with CO, including:
+1. **Transaction Re-execution**: One of the benefits of state sync is that it provides an
+   [intelligent syncing](https://aptos.dev/en/network/nodes/configure/state-sync#intelligent-syncing)
+   mode, where fullnodes can skip transaction re-execution and reuse the execution results from validators.
+   This helps to reduce CPU usage and improve synchronization time. However, CO does not
+   provide this as it requires fullnodes to re-execute blocks in real-time. As a result, it requires all fullnodes
+   to have equivalent (or better) hardware resources than validators, to ensure they can process blocks at the
+   same rates while dealing with other overheads (e.g., REST API requests).
+      - **Mitigation**: While this is a drawback, it does not pose a significant risk, as fullnodes are already
```

Comment:
Can a fullnode choose to not use CO, accept the latency penalty at high load, and potentially save hardware cost if they want to? or is this going to be an enforced sync mode for all nodes?

Reply:
Yes, it's config gated, so nodes can opt-out and use regular state sync if they want. I'll update the text 😄